### PR TITLE
applications: nrf_desktop: provide missing help for Kconfig options

### PR DESCRIPTION
**Pull request by emob-nordic**
As suggested in [comment](https://github.com/nrfconnect/sdk-nrf/pull/4951#discussion_r655268181) this PR fills missing help descriptions for Kconfig options in nrf_desktop application.

Jira: NCSDK-10210

### CODE REVIEW

File: applications/nrf_desktop/src/modules/Kconfig.config_channel [hunk 1]
```diff
@@ -34,6 +34,8 @@ config DESKTOP_CONFIG_CHANNEL_TIMEOUT
 	int "Transaction timeout on configuration channel in seconds"
 	depends on DESKTOP_CONFIG_CHANNEL_ENABLE
 	default 10
+	help
+	  Timeout in [s] after which config channel transaction is dropped.
```
Comment (MarekPieta) [inline applications/nrf_desktop/src/modules/Kconfig.config_channel:38]: Maybe `Timeout in seconds` or `Timeout [s]`?

File: applications/nrf_desktop/src/events/Kconfig [hunk 2]
```diff
@@ -11,72 +11,104 @@ menu "Event options"
...
-	bool "Battery level event"
+	bool "Log battery level events"
 	default y
+	help
+	  Log battery level events in nRF Desktop application.
 
 config DESKTOP_INIT_LOG_BLE_DISC_COMPLETE_EVENT
-	bool "BLE discovery complete event"
+	bool "Log Bluetooth LE discovery complete events"
 	default y
```
Comment (MarekPieta) [inline applications/nrf_desktop/src/events/Kconfig:27]: Help is missing

File: applications/nrf_desktop/src/modules/Kconfig.profiler_sync [hunk 3]
```diff
@@ -30,21 +30,31 @@ choice
...
+	help
+	  This option defines which gpio port is to be used to synchronize
+	  timestamps between two different devices.
 
 config DESKTOP_PROFILER_SYNC_GPIO_PIN
 	int "Sync GPIO pin"
 	range 0 31
 	default 0
+	help
+	  This option defines which gpio pin is to be used to synchronize
```
Comment (MarekPieta) [inline applications/nrf_desktop/src/modules/Kconfig.profiler_sync:56]: to generate/receive synchronization signal?

Comment (MarekPieta) [reply applications/nrf_desktop/src/modules/Kconfig.profiler_sync:56]: Same for port

File: applications/nrf_desktop/src/events/Kconfig (hunk 2 above)
Comment (emob-nordic) [reply applications/nrf_desktop/src/events/Kconfig:27]: Must have remove it by mistake, thanks - will add it.

**Review by MarekPieta — APPROVED** [review]

**Comment by MarekPieta** [comment]
I still wonder if we need this Kconfig help everywhere (both in menuconfig and in generated documentation it's just a duplicated text).

Maybe we could consider limiting the change only to labels of  the initially logged events? Currently they may be misleading in generated documentation (in menuconfig additional context is provided as comment).

**Review by pdunaj — APPROVED** [review]

**Pull request merged**